### ISSUE TITLE
HDF5 backend error in H5Dget_space

### ISSUE THREAD

**Issue by jhendersonHDF** [hpc/ior#428]
When running IOR with HDF5 as in:

`ior -a=HDF5 -t=1m -i=2 -r -R -w -W`

or

``ior -a=HDF5 -t=1m -i=2 -R -w -W``

I get the following:

```
HDF5-DIAG: Error detected in HDF5 (1.13.2-1) MPI-process 0:
  #000: /home/jhenderson/git/hdf5/src/H5D.c line 605 in H5Dget_space(): unable to synchronously get dataspace
    major: Dataset
    minor: Can't get value
  #001: /home/jhenderson/git/hdf5/src/H5D.c line 565 in H5D__get_space_api_common(): invalid dataset identifier
    major: Invalid arguments to routine
    minor: Inappropriate type
```

```
** error **
ERROR in aiori-HDF5.c (line 571): cannot get data space from data set.
** exiting **
```

It appears to be happening because IOR calls the `backend->close` routine at the end of the checkWrite block, which ends up closing the dataset reference that the HDF5 backend is holding on to. Then, the call to `WriteOrRead` in the checkRead block can't operate correctly, because the HDF5 backend doesn't have a valid dataset reference.

I experimented a bit, but couldn't come up with a good solution since simply delaying the dataset close isn't enough (the file can't be closed in the checkWrite block when objects in the file are still open) and the alternative appeared to be rearchitecting the `firstReadCheck/startNewDataSet` code in `HDF5_Xfer`. It's possible that particular code may have drifted out of sync with IOR development, since it says the readcheck operation will pass through that function twice, but I only see 1 invocation each pass.

Note also that the transfer size setting of `-t=1m` seems to be important for some reason. If I don't explicitly set it, I don't see the failure.

**Comment by JulianKunkel** [comment]
Thanks for reportign. I confirm the bug. It only happens with -i > 1. I submitted a PR with the fix.
Will close this issue, please check in the PR #429

**Comment by jhendersonHDF** [comment]
Can confirm the issue's fixed for me in #429. Thanks!
